bundles/org.openhab.binding.modbus.stiebeleltron/src/main/java/org/openhab/binding/modbus/stiebeleltron/internal/handler/StiebelEltronHandler.java

```diff
@@ -0,0 +1,713 @@
+import org.openhab.binding.modbus.stiebeleltron.internal.parser.SystemParameterBlockParser;
+import org.openhab.binding.modbus.stiebeleltron.internal.parser.SystemStateBlockParser;
+import org.openhab.io.transport.modbus.AsyncModbusFailure;
+import org.openhab.io.transport.modbus.ModbusCommunicationInterface;
+import org.openhab.io.transport.modbus.ModbusReadFunctionCode;
+import org.openhab.io.transport.modbus.ModbusReadRequestBlueprint;
+import org.openhab.io.transport.modbus.ModbusRegister;
+import org.openhab.io.transport.modbus.ModbusRegisterArray;
+import org.openhab.io.transport.modbus.ModbusWriteRegisterRequestBlueprint;
+import org.openhab.io.transport.modbus.ModbusWriteRequestBlueprint;
+import org.openhab.io.transport.modbus.PollTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Modbus.StiebelEltronHandler} is responsible for handling commands,
+ * which are sent to one of the channels and for polling the modbus.
+ *
+ * @author Paul Frank - Initial contribution
+ */
+@NonNullByDefault
+public class StiebelEltronHandler extends BaseThingHandler {
+
+    public abstract class AbstractBasePoller {
+
+        private final Logger logger = LoggerFactory.getLogger(StiebelEltronHandler.class);
+
+        private volatile @Nullable PollTask pollTask;
+
+        public synchronized void unregisterPollTask() {
+            PollTask task = pollTask;
+            if (task == null) {
+                return;
+            }
+
+            ModbusCommunicationInterface mycomms = StiebelEltronHandler.this.comms;
+            if (mycomms != null) {
+                mycomms.unregisterRegularPoll(task);
+            }
+            pollTask = null;
+        }
+
+        /**
+         * Register poll task This is where we set up our regular poller
+         */
+        public synchronized void registerPollTask(int address, int length, ModbusReadFunctionCode readFunctionCode) {
+
+            logger.debug("Setting up regular polling");
+
+            ModbusCommunicationInterface mycomms = StiebelEltronHandler.this.comms;
+            StiebelEltronConfiguration myconfig = StiebelEltronHandler.this.config;
+            if (myconfig == null || mycomms == null) {
+                throw new IllegalStateException("registerPollTask called without proper configuration");
+            }
+
+            ModbusReadRequestBlueprint request = new ModbusReadRequestBlueprint(getSlaveId(),
+                    readFunctionCode, address, length, myconfig.getMaxTries());
+
+
+            long refreshMillis = myconfig.getRefreshMillis();
+
+            pollTask = mycomms.registerRegularPoll(request, refreshMillis, 1000, result -> {
+                result.getRegisters().ifPresent(this::handlePolledData);
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            }, failure -> {StiebelEltronHandler.this.handleReadError(failure);});
```

Comment:
Syntactical sugar. Same for below.
```suggestion
            }, StiebelEltronHandler.this::handleReadError);
```

Reply:
Changed to your proposal